ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,453 @@
+version 1.0
+
+workflow TestingNewImputation {
+  input {
+    Int chunkLength = 25000000
+    Array[File] array_vcfs
+    Array[File] array_vcf_indices 
+    Array[String] samples
+    Boolean perform_qc_steps
+    File ref_dict = "gs://gcp-public-data--broad-references/hg19/v0/Homo_sapiens_assembly19.dict"
+    String genetic_maps_eagle = "/genetic_map_hg19_withX.txt.gz"
+    String output_callset_name = "broad_imputation"
+    String path_to_reference_panel = "gs://broad-dsde-methods-skwalker/polygenic_risk_scores/eagle_reference_panels/"
+    String path_to_m3vcf = "gs://broad-dsde-methods-skwalker/polygenic_risk_scores/minimac3_files/"
+  }
+
+  call GenerateDataset {
+    input:
+      input_vcfs = array_vcfs,
+      input_vcf_indices = array_vcf_indices,
+      output_vcf_basename = "merged_aou_arrays"
+  }
+
+  if (perform_qc_steps) {
+        call QConArray {
```

Comment:
I'm thinking this task should be combined with QC on Chunk or at least called on the chunk level instead (it'll be faster / less expensive this way ) -- @kachulis , @edytamalolepsza: thoughts?

Reply:
I would lean toward keeping QConChunk and QConArray separate, since they are really doing pretty different thing (one removing individual sites, the other deciding if a particular chunk of genome can be phased/imputed).  But I do agree that moving this into the scattered part of the pipeline could be a nice speedup.  I actually don't think it will make things less expensive though, since you still need to do the same amount of computation, just scattered across a number of machines in parallel.  It might even be slightly more expensive since there will be duplication of overhead, but probably worth it if this is a speed bottleneck.